Dynatrace/dynatrace-operator
# Align ProcessModuleConfig secret reconciler with other reconcilers

## Description

This PR refactors the process module config secret reconciler to use a Query object. 
It as well aligns the structure with other reconcilers and adds a cleanup for the secret created.

[Jira](https://dt-rnd.atlassian.net/browse/DAQ-1455)



## How can this be tested?

- Create a CloudNativeFullStack DynaKube and check if the pmc secret exists.
- Check the conditions in the DynaKube for pmc as well
